requirements/dev

```diff
@@ -0,0 +1,5 @@
+-r ./core
```

Comment:
nice change! We could also add all of the `pytest` stuff here?

Reply:
Hm...I guess that might mess with the github workflows stuff though... So nevermind? Unless you think it would also be good to split things up and have a separate file for testing dependencies that also gets installed in the workflows so we can run tests in the ci pipeline.

Reply:
True, yeah we could probably make that happen

Reply:
Seems fine adding pytest and related stuff to dev requirements?

Had to change the job to use the dev requirements, but don't think it matters if the jobs have the dev requirements (vs. just the core requirements)

Reply:
Yeah that seems good!